apps/nextjs-x-chain/src/app/page.tsx

```diff
@@ -272,34 +253,40 @@ function WalletConnection({
           />
         </div>
 
-        {(wallet instanceof SolanaDerivedWallet ||
-          wallet instanceof EIP1193DerivedWallet) && (
-          <div className="flex flex-col gap-6">
-            <h4 className="text-lg font-medium">Original Account Info</h4>
-            <LabelValueGrid
-              items={[
-                {
-                  label: "Address",
-                  value: (
-                    <DisplayValue
-                      value={originWalletDetails?.address ?? "Not Present"}
-                      isCorrect={!!originWalletDetails?.address}
-                    />
-                  ),
-                },
-                {
-                  label: "Public key",
-                  value: (
-                    <DisplayValue
-                      value={originWalletDetails?.publicKey ?? "Not Present"}
-                      isCorrect={!!originWalletDetails?.publicKey}
-                    />
-                  ),
-                },
-              ]}
-            />
-          </div>
-        )}
+        {wallet &&
+          (isSolanaDerivedWallet(wallet) || isEIP1193DerivedWallet(wallet)) && (
+            <div className="flex flex-col gap-6">
+              <h4 className="text-lg font-medium">Original Account Info</h4>
+              <LabelValueGrid
+                items={[
+                  {
+                    label: "Address",
+                    value: (
+                      <DisplayValue
+                        value={
+                          originWalletDetails?.address.toString() ??
+                          "Not Present"
+                        }
+                        isCorrect={!!originWalletDetails?.address}
+                      />
+                    ),
+                  },
+                  {
+                    label: "Public key",
+                    value: (
+                      <DisplayValue
+                        value={
+                          originWalletDetails?.publicKey?.toString() ??
+                          "Not Present"
+                        }
+                        isCorrect={!!originWalletDetails?.publicKey}
+                      />
+                    ),
+                  },
+                ]}
+              />
+            </div>
```

Comment:
how would you feel about something like:

```typescript
if (isSolanaDerivedWallet(wallet)) {
  // show solana-specific UI
} else if (isEIP1193DerivedWallet(wallet) {
  // show etherem-specific UI
}
```

the reason being, "public keys" are not applicable to ethereum, and the public key "is" the address for Solana
not sure if it's "useful" to try and combine them into the same data type

Reply:
Or alternatively, just show the address :)

Reply:
good call. maybe just display it in the "wallet info" section